### PR TITLE
Feature/1456 20160902 phoebe current addrafts

### PR DESCRIPTION
**Pull request by elia-ori**
### Tick ::

https://trello.com/c/dFBjvB1a/1456-current-addrafts
### Description ::

監聽「https://graph.facebook.com/*」  
拿到「current-addrafts」的相關資料  
然後分割「token」、「businessId」、「adAccountId」三個參數

### CODE REVIEW

**Comment by PhoebePan** [comment]
+1

File: background.js [hunk 1]
```diff
@@ -1,23 +1,41 @@
...
 chrome.tabs.onUpdated.addListener(checkUrl);
-console.log("!!! background")
 
+chrome.webRequest.onBeforeSendHeaders.addListener(function(details) {
+
+  if(details.url.match("current_addrafts")){
+    var params =  details.url.split("/");
+    var token = params[5].split("&")[0].replace("current_addrafts?access_token=","");
+    var businessId = params[5].split("&")[2].replace("__business_id=","");
+    var adAccountId = params[4].replace("act_","");
```
Comment (malikid) [inline background.js]: ### Line 27 - 31

還是要 follow coding style 喔~

File: background.js [hunk 2]
```diff
@@ -1,23 +1,41 @@
...
+
+    urlParams.token = token;
+    urlParams.businessId = businessId;
+    urlParams.adAccountId = adAccountId;
+
+    // todo: send message to content js for AA Miner API
+    console.log(urlParams);
+  }
+
+}, {urls: ["<all_urls>"]}, ["blocking", "requestHeaders"]);
```
Comment (malikid) [inline background.js]: ### Line 41

##### Usage Example

```
var callback = function(details) {...};
var filter = {...};
var opt_extraInfoSpec = [...];

chrome.webRequest.onBeforeRequest.addListener(callback, filter, opt_extraInfoSpec);
```

##### blocking

> **If the optional opt_extraInfoSpec array contains the string 'blocking' (only allowed for specific events), the callback function is handled synchronously. That means that the request is blocked until the callback function returns.**
> 
> In this case, the callback can return a webRequest.BlockingResponse that determines the further life cycle of the request. Depending on the context, this response allows cancelling or redirecting a request (onBeforeRequest), cancelling a request or modifying headers (onBeforeSendHeaders, onHeadersReceived), or providing authentication credentials (onAuthRequired).

##### requestHeaders

才能拿到 request 的 header 內容。

Comment (malikid) [reply background.js]: ### Line 41

建議 `{urls: ["<all_urls>"]}` 可以調整為 `{urls: ["*://graph.facebook.com/*/current_addrafts*"]}` 可以較省資源，不確定是否可辦到，可以試試~

Comment (PhoebePan) [reply background.js]: try 成功！！！ 可侷限 "_://graph.facebook.com/_/current_addrafts*"，感謝！！！

File: background.js [hunk 3]
```diff
@@ -1,23 +1,36 @@
...
 };
 
 // 只要 tab 更新就會在背景觸發比對網址是不是 "business.facebook.com"，決定要不要啟動 AAminer
 chrome.tabs.onUpdated.addListener(checkUrl);
-console.log("!!! background")
 
+chrome.webRequest.onBeforeSendHeaders.addListener(function(details) {
+  var params = details.url.split("/");
+  var token = params[5].split("&")[0].replace("current_addrafts?access_token=", "");
+  var businessId = params[5].split("&")[2].replace("__business_id=", "");
```
Comment (malikid) [inline background.js]: ### Line 27 - 28

建議把 `params[5].split("&")` 提出~

**Pull request merged**